compiler/src/dotty/tools/dotc/inlines/PrepareInlineable.scala

```diff
@@ -86,6 +89,11 @@ object PrepareInlineable {
 
       override def transform(tree: Tree)(using Context): Tree =
         postTransform(super.transform(preTransform(tree)))
+
+      protected def checkUnstableAccessor(accessedTree: Tree, accessor: Symbol)(using Context): Unit =
+        if ctx.settings.WunstableInlineAccessors.value then
+          val accessorTree = accessorDef(accessor, accessedTree.symbol)
```

Comment:
Something I'm now wondering: if this annotation doesn't in fact change the bytecode we generate, then why do we bother going through the inline accessor when calling non-annotated protected/package-private methods? Couldn't we always make a direct call even if the method isn't annotated?

Reply:
We need the annotation as a way to tell MiMa that this protected/package-private method can be used from outside the scope where it is defined.

We might be able to do this optimization once we have established `-WunstableInlineAccessors` and can break the old accessors. In this case we would still need to warn if that definition is not annotated with `@publicInBinary`.

Reply:
We could start using some of those package-private methods directly now but still generate the accessor. I will look into it after this PR and after the 3.4 release.